app/models.py

```diff
@@ -1642,6 +1644,86 @@ def serialize(self):
 
         return serialized
 
+    def serialize_with_cost_data(self):
+        serialized = self.serialize()
+        serialized["cost_details"] = {}
+        serialized["cost_in_pounds"] = "0.00"
+        serialized["is_cost_data_ready"] = True
+
+        if self.notification_type == "sms":
+            return self._add_cost_data_for_sms(serialized)
+        elif self.notification_type == "letter":
+            return self._add_cost_data_for_letter(serialized)
+
+        return serialized
+
+    def _add_cost_data_for_sms(self, serialized):
+        if not self._is_cost_data_ready_for_sms():
+            serialized["is_cost_data_ready"] = False
+            serialized["cost_details"] = {}
+            serialized["cost_in_pounds"] = None
+        else:
+            serialized["cost_details"]["billable_sms_fragments"] = self.billable_units
+            serialized["cost_details"]["international_rate_multiplier"] = self.rate_multiplier
+            sms_rate = self._get_sms_rate()
```

Comment:
are we worried about extra DB queries - should this be cached?

(i'm mostly wondering what happens if someone makes a get request to get the status of all notifications, or if we end up using this code in other bulk ways). 

might be something we just note in a TODO comment and do nothing about for now

Reply:
I was worried a little bit - I think it would be a good idea to cache it actually - what we'd do as a cache key, `created_at` cast to date (without time)?

And for letters we'd have to do the date, postage, and sheets, but also could work? I'm happy to add this today.

Just let's work together to keep review cycle short and get it out before I go on holiday hahah :D

Reply:
Following conversation:
- for now add get_cache for redis, check if it also automatically sets when value not there.
- in the future think about moving to serialized models

Reply:
Caching is added for both getting sms rate, and getting letter rates now.